tests/unit/plugins/mesos/execution_framework_test.py

```diff
@@ -1,54 +1,571 @@
+import threading
+import time
+
+import mock
 import pytest
+from addict import Dict
+from pyrsistent import v
+
+from task_processing.plugins.mesos import execution_framework as ef_mdl
+from task_processing.plugins.mesos import mesos_executor as me_mdl
+
 
-from task_processing.plugins.mesos.execution_framework import (
-    ExecutionFramework
-)
+@pytest.fixture
+def mock_get_metric():
+    with mock.patch.object(ef_mdl, 'get_metric') as mock_get_metric:
+        yield mock_get_metric
 
 
 @pytest.fixture
 def ef():
-    return ExecutionFramework("name", "role")
+    threading.Thread = mock.Mock()
```

Comment:
It does seem to work for me. pdb shows that the Thread method called in ExecutionFramework is a mock object. But I suppose using a fixture makes it cleaner?